server.js

```diff
@@ -1,29 +1,114 @@
 import express from 'express'
-import bodyParser from 'body-parser'
+import listEndpoints from 'express-list-endpoints'
 import cors from 'cors'
+import dotenv from 'dotenv'
 import mongoose from 'mongoose'
 
+dotenv.config()
+
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true
+})
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Message is required"],
+    unique: [true, "Message has to be unique"],
+    minlength: [5, "Message has to be at least 5 characters"],
+    maxlength: [140, "Message cannot exceed 140 characters"]
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// End point to POST a new thought
+app.post('/thoughts', async (req, res) => {
```

Comment:
I can't send any new message to your API :( it returns an error 400 every time
![image](https://user-images.githubusercontent.com/50926159/120110000-2a957a00-c16c-11eb-9bf5-45b0ed00d5ca.png)